### ISSUE TITLE
UnreachableCode: false positive when use error from Coil

### ISSUE THREAD

**Issue by osipxd** [detekt/detekt#4250]
## Observed Behavior

```kotlin
imageView.load(imageUrl) {
    placeholder(R.drawable.image_placeholder)
    error(R.drawable.image_error)       // This is not Kotlin's error function
    fallback(R.drawable.image_fallback) // UnreachableCode reported here, but this code is reachable
}
```

## Context

We use [Coil](https://github.com/coil-kt/coil) library for image loading and there are [methods called `error`](https://github.com/coil-kt/coil/blob/f6a71ddd79faaf3332bef9baa85f85b1a05863e2/coil-base/src/main/java/coil/request/ImageRequest.kt#L796-L810).
It looks like detekt considers all functions with the name `error` as Kotlin's `error` function.

## Steps to reproduce

[A project to reproduce the issue](https://github.com/osipxd/detekt-4250-unreachable-code)

## Your Environment

* Version of detekt used: 1.18.0

**Comment by BraisGabin** [comment]
I just created a fast Unit test and I can't reproduce your behaviour:

```kotlin
        it("Does not report unreachable code after custom error function") {
            val code = """
                fun error(a: Int) {
                    println(a)
                }

                fun test(): Int {
                    error(5)
                    return 0
                }
            """
            assertThat(subject.compileAndLintWithContext(env, code)).isEmpty()
        }
```

Could you provide a project to reproduce this issue?

**Comment by osipxd** [comment]
[Here](https://github.com/osipxd/detekt-4250-unreachable-code) is the simple reproducer project.

[This](https://github.com/osipxd/detekt-4250-unreachable-code/commit/b6f2abfb3d03a0acd018ea380c77e3f269042387) is the commit with code related to the issue, and [this](https://github.com/osipxd/detekt-4250-unreachable-code/runs/4155329209) is the failing task on CI.
